fhir-server-test/src/test/java/com/ibm/fhir/server/test/SearchReferenceTest.java

```diff
@@ -81,25 +94,27 @@ public String createPatientWithReference(String field, String reference) throws
     public void testCreatePatient() throws Exception {
         // references without an explicit type will be interpreted as literals (e.g. externals) not local references
         // we will not infer their type, which means that searches specifying a type will not match them by design
-        patientResourceIdWithReferenceId = createPatientWithReference("organization", "3001");
+        patientResourceIdWithReferenceId = createPatientWithReference("organization", "3001", null);
 
         // relative reference
-        patientResourceIdWithReferenceTypeId = createPatientWithReference("organization", "Organization/3002");
+        patientResourceIdWithReferenceTypeId = createPatientWithReference("organization", "Organization/3002", null);
 
         // relative reference because the url aligns with the server base address
-        patientResourceIdWithReferenceUrl = createPatientWithReference("organization", "https://localhost:9443/fhir-server/api/v4/Organization/3003");
+        patientResourceIdWithReferenceUrl = createPatientWithReference("organization", "https://localhost:9443/fhir-server/api/v4/Organization/3003", null);
 
-        // patientResourceIdWithReferenceCanonicalUri = createPatientWithReference("https://localhost:9443/fhir-server/api/v4/Organization/3004|1.0.0");
+        // patientResourceIdWithReferenceCanonicalUri = createPatientWithReference("https://localhost:9443/fhir-server/api/v4/Organization/3004|1.0.0", null);
 
         // Literal reference to another server
-        patientWithLiteralReference = createPatientWithReference("organization", "https://an.example.com/Organization/3004");
+        patientWithLiteralReference = createPatientWithReference("organization", "https://an.example.com/Organization/3004", null);
 
         // relative reference to Practitioner for general practitioner
-        patientResourceIdWithPractitionerReferenceTypeId = createPatientWithReference("general-practitioner", "Practitioner/3002");
+        patientResourceIdWithPractitionerReferenceTypeId = createPatientWithReference("general-practitioner", "Practitioner/3002", null);
 
         // relative reference to PractitionerRole for general practitioner
-        patientResourceIdWithPractitionerRoleReferenceTypeId = createPatientWithReference("general-practitioner", "PractitionerRole/3002");
+        patientResourceIdWithPractitionerRoleReferenceTypeId = createPatientWithReference("general-practitioner", "PractitionerRole/3002", null);
 
+        // logical reference
+        patientWithLogicalReference = createPatientWithReference("organization", null, Identifier.builder().system(Uri.of("https://an.example.com/OrgId")).value(string("3005")).build());
```

Comment:
I think I'd also like to see a test case for a reference that has both a literal reference value AND a logical reference.  Does it work?

Reply:
Test cases added.